geoh5py/objects/cell_object.py

```diff
@@ -69,6 +69,20 @@ def cells(self, cells: np.ndarray | list | tuple):
 
         self.workspace.update_attribute(self, "cells")
 
+    @property
+    def centroids(self) -> np.ndarray | None:
+        """
+        Compute the centroids of the cells.
+        """
+        if self.cells is None:
+            return None
+
+        return np.array([self.vertices[conn].mean(axis=0) for conn in self.cells])
+
+    @property
+    def locations(self):
+        return self.vertices
```

Comment:
Would be better defined on the parent `Points` class, although technically already handled by the `BaseObject`

Reply:
That's to avoid the issue with the cell object

Reply:
OK